File: solutions/c/darts/1/darts.c
#include "darts.h"
#include <math.h>

static const uint8_t inner_circle  = 10;
static const uint8_t middle_circle = 5;
static const uint8_t outer_circle  = 1;
static const uint8_t missed        = 0;

static inline float distance_from_center(coordinate_t impact) {
    return sqrtf(impact.x * impact.x + impact.y * impact.y);
}

uint8_t score(coordinate_t impact) {
    float distance = distance_from_center(impact);
    if (distance <= 1.0f)       return inner_circle;
    else if (distance <= 5.0f)  return middle_circle;
    else if (distance <= 10.0f) return outer_circle;
    else                        return missed;
}

File: solutions/elixir/darts/1/lib/darts.ex
defmodule Darts do

  def score({x, y}) do
    radius = :math.sqrt(x*x + y*y)
    case radius do
      r when r <= 1.0 -> 10
      r when r <= 5.0 -> 5
      r when r <= 10.0 -> 1
      _ -> 0
    end
  end
  
end
